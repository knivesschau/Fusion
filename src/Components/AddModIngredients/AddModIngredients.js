import React, { useState } from 'react';
import './AddModIngredients.css';

export default function AddModIngredients() {
    const [modIngredients, setModIngredients] = useState({ ingredient: [] });

    function createInputs() {
        return modIngredients.ingredient.map((ingredient, i) => 
            <li id="add-mod-ingredients" key={i}>
                <input
                    type="text"
                    id="mod_ingredients"
                    name="mod_ingredients"
                    value={ingredient || ""}
                    onChange={handleUpdate.bind(i)}/>

                <button type="button" id="remove-mod-ingredient" onClick={removeInput.bind({i})}>Remove</button>
            </li>
        );
    };
    
    function handleUpdate(e) {
        let newIngredients = [...modIngredients.ingredient];
        newIngredients[this] = e.target.value;

        setModIngredients({
            ingredient: newIngredients
        });
    };

    const addInput = () => {
        setModIngredients({
            ingredient: [...modIngredients.ingredient, ""]
        });
    };

    const removeInput = () => {
        let newIngredients = [...modIngredients.ingredient];
        newIngredients.splice(this, 1);

        setModIngredients({
            ingredient: newIngredients
        });
    };

    return (
        <>
        {createInputs()}
        <button type="button" id="add-mod-ingredient" onClick={addInput}>Add Ingredient</button>
        </>
    );
};
