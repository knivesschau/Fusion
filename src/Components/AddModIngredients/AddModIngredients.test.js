import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AddModIngredients from './AddModIngredients';

describe ('AddModIngredients Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AddModIngredients/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete AddModIngredients Component', () => {
        const wrapper = shallow(<AddModIngredients/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});