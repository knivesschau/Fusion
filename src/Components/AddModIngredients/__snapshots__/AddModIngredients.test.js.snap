// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddModIngredients Component Renders the complete AddModIngredients Component 1`] = `
<Fragment>
  <button
    id="add-mod-ingredient"
    onClick={[Function]}
    type="button"
  >
    Add Ingredient
  </button>
</Fragment>
`;
